maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -1438,6 +1446,33 @@ private MavenExecutionRequest populateRequest( CliRequest cliRequest, MavenExecu
         return request;
     }
 
+    private void disableInteractiveModeIfNeeded( final CliRequest cliRequest, final MavenExecutionRequest request )
+    {
+        CommandLine commandLine = cliRequest.getCommandLine();
+        if ( commandLine.hasOption( FORCE_INTERACTIVE ) )
+        {
+            return;
+        }
+
+        boolean runningOnCI = isRunningOnCI( cliRequest.getSystemProperties() );
+        if ( runningOnCI )
+        {
+            slf4jLogger.info( "Detected a CI build, because the environment variable CI equals \"true\". "
+                + "Disable this detection by removing that variable or adding --force-interactive." );
```

Comment:
What does mean that we detected CI builds?
maybe, some thing like:

```
Using non interactive build due to CI environment present.
```

Reply:
I agree! That is much clearer for the users. I will change it.